Remove FromAsync usage from SendPacketsAsync on Unix

stephentoub opened the pull request:
Also contributes to https://github.com/dotnet/runtime/issues/33417

wfurt reviewed (approved):
LGTM. 
We do not expect behavior difference, right?

stephentoub commented:
> We do not expect behavior difference, right?

Correct.

stephentoub committed:
Remove FromAsync usage from SendPacketsAsync on Unix

This call is in the Unix implementation, and the Unix implementation of DisconnectAsync just calls the synchronous Disconnect... so we can just call it directly and skip all the unnecessary layering.

stephentoub commented:
I decided my change was overkill.  All I really cared about right now was removing the use of FromAsync, and since Unix's DisconnectAsync is actually entirely synchronous, I just changed it to call the existing implementation.
